cognite/client/utils/_token_generator.py

```diff
@@ -0,0 +1,48 @@
+import datetime
+from typing import *
+
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+
+from cognite.client.exceptions import CogniteAPIKeyError
+
+
+class TokenGenerator:
+    def __init__(self, token_url: str, client_id: str, client_secret: str, scopes: List[str]):
+        self.token_url = token_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.scopes = scopes
+
+        if self.token_params_set():
+            self._generate_access_token()
```

Comment:
this will potentially cause a fairly obscure exception to go up to user code, right? Can we convert that to something more sensible?

Reply:
Usually this is either a deliberately vague 401 error or something very clear like 'Tenant ABC not found'. For those very useful errors informing someone that they have put the arguments in wrong, I would rather throw that error up to user code for the user to handle.

We could do something like catching all errors, and pushing the error message inside an error of our own; but I am not sure if that is more sensible. Do we do that elsewhere in this sdk?

Reply:
HTTP errors and such get converted to CogniteAPIError
`raise CogniteAPIKeyError("somethingsomething token error") from actualtokenexception` ?

Reply:
Better than catching all errors: how about catching `Oauth2Error`s, e.g. all of the expected errors that might come from calling `fetch_token` on `OAuth2Session`?